app/src/main/java/fr/free/nrw/commons/mwapi/ApacheHttpClientMediaWikiApi.java

```diff
@@ -457,11 +458,13 @@ public String wikidataCreateClaim(String entityId, String property, String snakt
 
     @Nullable
     @Override
-    public String wikidataAddLabels(String fileEntityId, String caption) throws IOException {
-        CustomApiResult result = api.action("wbeditentity")
+    public String wikidataAddLabels(String fileEntityId, Map<String, String> caption) throws IOException {
```

Comment:
@vanshikaarora  Why are we still adding APIs in `ApacheHttpClientMediaWikiApi` which uses XML and `AbstractHttpClient`. It will just add to our tech debt. We already have 29 APIs that needs to be migrated to Retrofit using the data-client library and @ashishkumar468 and I have been migrating all the APIs to 1 by 1. It would be great if we don't add any more APIs in  `ApacheHttpClientMediaWikiApi` and start using Retrofit. I remember posting the same comment earlier on one of your PRs. 

@nicolas-raoul @ashishkumar468 It would be great if you guys could help in making sure that we use the `data-client` as I might not be able to review all the diffs. :)